setup-build-env.sh

```diff
@@ -4,57 +4,56 @@ ensure_lxd() {
   if ! command -v lxc version &> /dev/null
   then
     echo "LXD could not be found. Please ensure LXD exists."
-    exit
+    exit 1
   fi
 }
 
 build_image_in_container() {
   
-  local OS_NAME="${OS_NAME:-ubuntu}"
-  local OS_VERSION="${OS_VERSION:-22.04}"
-  local ARCH="${ARCH:-$(uname -m)}"
   local IMAGE_ALIAS="${IMAGE_ALIAS:-${OS_NAME}-${OS_VERSION}-${ARCH}}"
 
-  local BUILD_PREREQS_PATH="${BUILD_PREREQS_PATH:-/home/ubuntu/remote-build/files}"
-  local DOTNET_SDK="${DOTNET_SDK:-dotnet-sdk-7.0.100-linux-ppc64le.tar.gz}"
-  local PATCH_FILE="${PATCH_FILE:-runner-ppc64le.patch}"
+  local BUILD_PREREQS_PATH="${SRCDIR}/build-files"
+  if [ ! -d "${BUILD_PREREQS_PATH}" ]; then
+      echo "Check the BUILD_PREREQS_PATH specification" >&2
+      exit 3
+  fi
+  local PATCH_FILE="${PATCH_FILE:-runner-${ARCH}.patch}"
 
   local BUILD_CONTAINER
   BUILD_CONTAINER="gha-builder-$(date +%s)"
-  lxc launch "${OS_NAME}:${OS_VERSION}" "${BUILD_CONTAINER}" 
+  lxc launch "${LXD_CONTAINER}" "${BUILD_CONTAINER}" 
   lxc ls
   
   # give container some time to wake up
-  sleep 10
+  sleep 5
```

Comment:
ooh, these can be soo tricky.  Is there a better mechanism?  and I'm a little concerned that Power may be slower under heavy load maybe than Z, or in either case these are probably all tested on a mostly idle machine.  Can we test in a slow loop if the container is up rather than "hoping" it will be ready in that time frame?

Reply:
See my other remark wrt sleeping.